components/data_comps/datm/bld/namelist_files/namelist_definition_datm.xml

```diff
@@ -4,32 +4,93 @@
 
 <namelist_definition>
 
-<!-- =========================================  -->
-<!--- DATM env settings                         -->
-<!-- =========================================  -->
+<!-- ========================================================================================  -->
+<!-- Stream file variables                                                                     -->
+<!-- ========================================================================================  -->
 
-<entry id="DATM_MODE"
-type="char*256"
-category="datm_setting"
-group="datm_env_settings" 
-valid_values="CLM_QIAN,CLM1PT,CLMCRUNCEP,CLMCRUNCEP_V5,CLMGSWP3,CPLHIST3HrWx,CORE2_NYF,CORE2_IAF,COPYALL_NPS_v1,COPYALL_CORE2_v1,WW3,NLDAS">
-Data Model mode to process forcing data
+<entry id="streamslist"
+type="char*1024(100)"
+category="streams"
+group="streams_file">
+List of streams used for the given datm_mode.
 </entry>
 
-<entry id="DATM_PRESAERO"
-type="char*256"
-category="datm_setting"
-group="datm_env_settings"
-valid_values="none,pt1_pt1,clim_2000,clim_1850,trans_1850-2000,rcp2.6,rcp4.5,rcp6.0,rcp8.5">
-Prescribed aerosol mode
+<entry id="strm_domdir"
```

Comment:
The xml schema was purposely not addressed in this implementation. I agree
with your suggestion - but I think an even more general schema that
combines both the namelist_definition.xml and namelist_defaults.xml file
needs to be created. That will be done in the next phase the project.

On Thu, Aug 11, 2016 at 7:38 AM, jedwards4b <notifications@github.com>
wrote:

> In components/data_comps/datm/bld/namelist_files/namelist_
> definition_datm.xml
> <https://github.com/ESMCI/cime/pull/377#discussion_r74423004>:
>
> >  </entry>
> >
> > -<entry id="DATM_PRESAERO"
> > -type="char*256"
> > -category="datm_setting"
> > -group="datm_env_settings"
> > -valid_values="none,pt1_pt1,clim_2000,clim_1850,trans_1850-2000,rcp2.6,rcp4.5,rcp6.0,rcp8.5">
> > -Prescribed aerosol mode
> > +<entry id="strm_domdir"
>
> I have a philosophical issue with the format of this xml file. The issue
> is in the distinction between attributes and entries: this file incorrectly
> handles everything as attributes. It would be more correct and consistent
> with other xml in cime to have:
>
>     <entry id="strm_domdir">
>         <type>char*4096</type>
>         <category>streams</category>
>         <group>streams_file</group>
>         <desc>Stream domain file directory</desc>
>     </entry>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ESMCI/cime/pull/377/files/8221c3b6b655fad45d3cb227136384e3407cb8db#r74423004>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHlxE9UeJVbuG14Wl3clIFKmeu5Nd1pnks5qeyXvgaJpZM4Jhszo>
> .
>